lib/ui.rb

```diff
@@ -0,0 +1,28 @@
+require_relative 'messages'
+
+class UserInterface
+    def initialize
+        @message = Message.new
+    end
+
+    def start_game
+        puts @message.lookup(:welcome)
+        puts @message.lookup(:instructions)
+        puts display_board
+    end
+
+    def get_spot
+        puts @message.lookup(:get_spot)
+        spot = gets.chomp.to_i
+        if valid_spot?(spot)
+            return spot
+        else
+            puts @message.lookup(:invalid_input)
+            get_spot
+        end
+    end
+
+    def valid_spot?(spot)
```

Comment:
I see you have a test for `get_spot` but not directly testing `valid_spot?`.  If you're only planning to use `valid_spot?` in `ui.rb`, then let's indicate this is a private function, and therefore you don't need to write a test for it directly (as you're only using it within get_spot).